### ISSUE TITLE
`amount` query parameter

### ISSUE THREAD

**Issue by mattroush** [CUCentralAdvancement/giving-frontend#570]
`amount` is one of [the custom query parameters](https://github.com/CUCentralAdvancement/giving-frontend/blob/main/docs/custom-query-params.md) users can leverage, especially in cases where a campaign promotes giving at a certain amount or where marketing automation platforms are able to generate unique links per recipient based on donor's giving history.

On fund pages only, the site enables "Suggested Amount" (*dev task TBD*) in the Add To Basket form, and the amount is populated by param value. If a web fund manager has already enabled Suggested Amount in fund settings, the amount is overwritten with the param value.

**Use Case:** Solicit class of 2022 to make a gift of $20.22 to the alumni scholarship fund.

**Validation:**
- can be any whole number or float (2 decimal places)
- should be less than 25000 (current per-transaction limit)
- cannot contain commas or special characters
✅ : `amount=20.22`
❌ :  `amount=$20`

### Acceptance Criteria

While the page is a fund page and param is present in URI...
- [x] site enables the "Suggested Amount" option if it is not already 
  - https://giving-frontend.herokuapp.com/fund/cancer-research-general-fund?amount=20.00
- [x] site sets the Suggested Amount value to the param's value, overriding the existing amount if present 
  - https://giving-frontend.herokuapp.com/fund/bridge-forward-scholarship?amount=40.00 
- [x] site sets "Suggested Amount" as the default amount selection 
  - Verified by clicking "Give Now" and seeing the cart show suggested amount value.
- [x] if param value is empty, do not execute 
  - https://giving-frontend.herokuapp.com/fund/bridge-forward-scholarship?amount= 
- [x] Once the suggested amount gets loaded, make sure you can click other values and enter an amount that gets sent to the cart. The suggested amount should not remain selected once the user takes those actions.
- [ ] Strip characters around the float value, e.g. ?amount=$20 is treated like ?amount=20. Probably best to only strip known characters, with `$` being the most obvious prefix to look for.
- [ ] Strip comma separators from value, e.g. ?amount=1,000 treated like ?amount=1000
- [ ] If "illegal" character(s) are found in the float...or other errors, then ignore and do nothing. 
- [ ] If the param value is malformed then ignore and do nothing.
- [ ] If the param value is above 25000, then ignore and do nothing.
- [ ] If the param value is below x.xx, then round up to x.xx?

### Next Issue

Deploy and test.

**Comment by alexfinnarn** [comment]
@mattroush a lot of this work was done in #440 but still some cleanup and edge cases. I added a check item:

>  If param value is malformed, then...either try to correct it, or more safely, just ignore the value.

But should some correction occur for something like https://giving-frontend.herokuapp.com/fund/bridge-forward-scholarship?amount=2.55555 ?

**Comment by mattroush** [comment]
@alexfinnarn I updated them. I like the idea of stripping known characters. "$" definitely, and added a new checkbox in to address comma separators (ex. 1,000.00).

For most of the others, I just changed to "ignore and do nothing" -- the users visiting the links with query param aren't the ones creating those links, and so an error message about a failed amount won't make any sense to them. As for the $25000+ limit, could just be user error -- user meant it to be 3000.00 but forgot the period -- but still, in that scenario the donor wouldn't know any amount was supposed to appear, let alone $24,999. I could see how showing an error might be helpful to folks who test their links beforehand, but not really an option when doing dynamic link creation with thousands of contacts in an email send and so best to take donor's perspective on showing errors (more confusing than helpful).

As for that last one -- rounding up if value has more than two decimals -- I feel like if a value has more than two, then it's more likely to be user error (accidentally hit the period instead of comma or something), in which case rounding doesn't actually solve the problem. Open to what you think there, and don't think it helps or hurts much either way.
